lib/sqlalchemy/sql/elements.py

```diff
@@ -794,16 +822,50 @@ def reverse_operate(
         ) -> ColumnElement[Any]:
             ...
 
+        @overload
         def op(
             self,
             opstring: str,
             precedence: int = 0,
             is_comparison: bool = False,
-            return_type: Optional[_TypeEngineArgument[_OPT]] = None,
+            *,
+            return_type: _TypeEngineArgument[_OPT],
+            python_impl: Optional[Callable[..., Any]] = None,
+        ) -> Callable[[Any], BinaryExpression[_OPT]]:
+            ...
+
+        @overload
+        def op(
+            self,
+            opstring: str,
+            precedence: int,
+            is_comparison: bool,
+            return_type: _TypeEngineArgument[_OPT],
             python_impl: Optional[Callable[..., Any]] = None,
         ) -> Callable[[Any], BinaryExpression[_OPT]]:
             ...
```

Comment:
these overloads dont seem to be correct, or I'm not following

AFAICT there should be two overloads:

1. is_comparison: True  -> return type is `Callable[[Any], BinaryExpression[bool]]`
2. return_type: `_TypeEngineArgument[_OPT]`  -> return type is `Callable[[Any], BinaryExpression[_OPT]]`
3. default return type is `Callable[[Any], BinaryExpression[Any]]`

Reply:
OK as it stands no difference in mypy can be produced, only pyright, but the problem is I dont understand this overload so i have to keep looking at it and figure some way to clarify this

Reply:
OK I have it now thanks